subprojects/docs/src/samples/userguide/java/basic/kotlin/build.gradle.kts

```diff
@@ -0,0 +1,106 @@
+// tag::apply-java-plugin[]
+plugins {
+    java
+}
+
+project.setProperty("sourceCompatibility", "1.8")
+project.setProperty("targetCompatibility", "1.8")
+
+version = "1.2.1"
+// end::apply-java-plugin[]
+
+// tag::java-dependency-mgmt[]
+repositories {
+    mavenCentral()
+}
+
+dependencies {
+    implementation("org.hibernate:hibernate-core:3.6.7.Final")
+}
+// end::java-dependency-mgmt[]
+
+// tag::java-basic-test-config[]
+dependencies {
+    testImplementation("junit:junit:4.12")
+}
+
+tasks.getByName<Test>("test") {
+    useJUnit()
+
+    maxHeapSize = "1G"
+}
+// end::java-basic-test-config[]
+
+// tag::practical-integ-test-source-set[]
+sourceSets {
+    create("intTest") {
+        compileClasspath += sourceSets["main"].output
+        runtimeClasspath += sourceSets["main"].output
+    }
+}
+
+val intTestImplementation by configurations.getting {
+    extendsFrom(configurations.implementation)
+}
+
+configurations["intTestRuntimeOnly"].extendsFrom(configurations.runtimeOnly)
+
+dependencies {
+    intTestImplementation("junit:junit:4.12")
+}
+// end::practical-integ-test-source-set[]
+
+// tag::using-custom-doclet[]
+val asciidoclet by configurations.creating
+
+dependencies {
+    asciidoclet("org.asciidoctor:asciidoclet:1.+")
+}
+
+tasks.getByName<Javadoc>("javadoc") {
+    options.docletpath = asciidoclet.files.toList()
+    options.doclet = "org.asciidoctor.Asciidoclet"
+}
+// end::using-custom-doclet[]
+
+
+// tag::skip-tests-condition[]
+tasks["test"].onlyIf { !project.hasProperty("mySkipTests") }
+// end::skip-tests-condition[]
+
+// tag::java-compiler-options[]
+tasks.getByName<JavaCompile>("compileJava") {
+    options.apply {
+        isIncremental = true
+        isFork = true
+        isFailOnError = false
+    }
+}
+// end::java-compiler-options[]
+
+// tag::integ-test-task[]
+val integrationTest by tasks.creating(Test::class) {
```

Comment:
I think for this case we want to prefer the `val integrationTest = task("integrationTest", Test::class) {}`

Reply:
I think `val integrationTest = tasks.create<Test>("integrationTest")` is cleaner. It shows that the intent is to *create* a task, and avoids the use of `Test::class`. It's also easier to migrate to

```
tasks {
    val integrationTest = create<Test>("integrationTest") {}
    ...
```

which seems to be the preferred option when creating/configuring several tasks.